provider/openstack/provider.go

```diff
@@ -1512,6 +1519,47 @@ func (e *Environ) networksForModel() ([]nova.ServerNetworks, error) {
 	return append(networks, nova.ServerNetworks{NetworkId: networkID}), nil
 }
 
+// useNetwork returns the name of the network to use.
+// First check the model config.  If that is empty, look
+// for internal networks.  If there is more than one, error.
+// If there is only one, use it and set the network model
+// config to use it in the future.
+func (e *Environ) useNetwork() (string, error) {
+	usingNetwork := e.ecfg().network()
+	if usingNetwork != "" {
+		return usingNetwork, nil
+	}
+	var err error
+	usingNetwork, err = e.findOneInternalNetwork()
+	if err != nil {
+		return "", err
+	}
+	cfg, err := e.ecfg().Apply(map[string]interface{}{NetworkKey: usingNetwork})
+	if err != nil {
+		return "", err
+	}
+	if err := e.SetConfig(cfg); err != nil {
+		return "", errors.Annotatef(err, "updating network in config")
+	}
+	logger.Warningf("no network provided, only one internal network found, added network %q to model-config", usingNetwork)
```

Comment:
This sets the value on the Environ instance, how does it get to be saved in state?

Reply:
It appears that part of the bootstrap process took the updated config and saved it to state.   Unfortunately the same does not happen to other models.  The value is found, but not saved to state, nor seen in model-config.  

We decided to annotate the error message if bootstrap fails due to an unconfigured network and none are found to improve the user experience.  Getting data out of an environ and into state, or even the model-default config is not straightforward at this time.  Therefore it's a stretch goal.